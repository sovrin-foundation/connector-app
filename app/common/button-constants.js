// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
