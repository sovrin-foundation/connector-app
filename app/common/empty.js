// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
