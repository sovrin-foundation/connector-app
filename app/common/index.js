// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
