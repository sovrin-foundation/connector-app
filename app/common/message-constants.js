// @flow
// for overall app message string, it can be error, success, warning etc.

export const TOUCH_ID_MESSAGE = 'Please confirm with Biometrics'
export const TOUCH_ID_NOT_AVAILABLE =
  'Biometrics is not supported on this device'

export const DEVICE_ENROLLMENT_ERROR = [
  'Enrollment Error',
  'Identifier or Seed or Push-notification token not present!',
]

export const PUSH_NOTIFICATION_PERMISSION_ERROR = [
  'Permission Error',
  'Push Notification permission needed! Please enable it from Settings.',
]

export const DUPLICATE_CONNECTION_ERROR = [
  'Attention!',
  'Duplicate Connection invitation!',
]

export const UNLOCKING_APP_WAIT_MESSAGE = 'Loading please wait...'

export const ENTER_PASS_CODE_MESSAGE = 'Enter passcode'

export const ENTER_YOUR_PASS_CODE_MESSAGE = 'Enter your passcode'

export const FIRST_BACKUP_TITLE = 'Set up data recovery now!'

export const FIRST_BACKUP_SUBTEXT = '2 easy steps, takes 3 minutes.'

export const SUBSEQUENT_BACKUP_TITLE =
  'You have new data that is not backed up.'

export const SUBSEQUENT_BACKUP_SUBTEXT =
  "Create another backup so you don't lose anything."

export const PASSPHRASE_GENERATION_ERROR =
  'Uh oh... Looks like something went wrong.'
