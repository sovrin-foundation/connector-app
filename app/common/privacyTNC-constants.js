// @flow

import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const localPrivacyPolicySource = isAndroid
  ? 'file:///android_asset/external/connectme/privacy.html'
  : './privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
