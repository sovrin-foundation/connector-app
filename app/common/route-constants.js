// @flow
export const splashScreenRoute = 'SplashScreenHide'
export const homeRoute = 'Home'
export const authenticationRoute = 'Authentication'
export const connectionRoute = 'Connection'
export const settingsRoute = 'Settings'
export const settingsScreen = 'SettingsStack'
export const expiredTokenRoute = 'ExpiredToken'
export const qrCodeScannerRoute = 'QRCodeScanner'
export const lockPinSetupRoute = 'PinSetup'
export const lockTouchIdSetupRoute = 'TouchIdSetup'
export const lockPinSetupHomeRoute = 'PinSetupHome'
export const lockSelectionRoute = 'LockSelection'
export const lockEnterPinRoute = 'LockEnterPin'
export const lockSetupSuccessRoute = 'LockSetupSuccess'
export const lockEnterFingerprintRoute = 'LockEnterFingerprint'
export const claimOfferRoute = 'ClaimOffer'
export const homeTabRoute = 'TabHome'
export const connectionsTabRoute = 'TabHome'
export const credentialsTabRoute = 'TabCredentials'
export const discoverTabRoute = 'TabDiscover'
export const menuTabRoute = 'TabMenu'
export const settingsTabRoute = 'TabSettings'
export const qrCodeScannerTabRoute = 'TabQRCodeScanner'
export const proofRequestRoute = 'ProofRequest'
export const invitationRoute = 'InvitationRoute'
export const connectionHistoryRoute = 'ConnectionHistoryRoute'
export const connectionHistoryDetailsRoute = 'ConnectionHistoryDetailsRoute'
export const switchEnvironmentRoute = 'SwitchEnvironment'
export const lockAuthorizationRoute = 'LockAuthorizationRoute'
export const lockAuthorizationHomeRoute = 'LockAuthorizationHomeRoute'
export const waitForInvitationRoute = 'WaitForInvitationRoute'
export const connectionHistoryPendingRoute = 'ConnectionHistoryPendingRoute'
export const walletRoute = 'TokenWallet'
export const walletTabSendDetailsRoute = 'WalletTabSendDetails'
export const receiveTabRoute = 'WalletReceiveTab'
export const sendTabRoute = 'WalletSendTab'
export const historyTabRoute = 'WalletHistoryTab'
export const aboutAppRoute = 'AboutApp'
export const onfidoRoute = 'Onfido'
export const privacyTNCRoute = 'PrivacyTNC'
export const eulaRoute = 'EulaScreen'
export const genRecoveryPhraseRoute = 'GenerateRecoveryPhrase'
export const verifyRecoveryPhraseRoute = 'VerifyRecoveryPhrase'
export const exportBackupFileRoute = 'ExportBackupFile'
export const backupCompleteRoute = 'BackupComplete'
export const restoreRoute = 'RestoreStartScreen'
export const restoreWaitRoute = 'RestoreWaitScreen'
export const restorePassphraseRoute = 'RestorePassphrase'
export const backupErrorRoute = 'BackupErrorScreen'
