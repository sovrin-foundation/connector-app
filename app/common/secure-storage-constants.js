// @flow

export const PUSH_COM_METHOD = 'pushComMethod'
export const CONNECT_ME_STORAGE_KEY = '@connectmeDefaults'
export const IS_ALREADY_INSTALLED = `${CONNECT_ME_STORAGE_KEY}:IS_ALREADY_INSTALLED`
export const CONNECTIONS = 'CONNECTIONS'
export const CLAIM_MAP = `CLAIM_MAP`
export const WALLET_ENCRYPTION_KEY = 'walletEncryptionKey'
export const WALLET_BALANCE = 'walletBalance'
export const WALLET_ADDRESSES = 'walletAddresses'
export const WALLET_HISTORY = 'walletHistory'
export const LAST_SUCCESSFUL_BACKUP = 'lastSuccessfulBackup'
export const PASSPHRASE_SALT_STORAGE_KEY = 'passphraseSalt'
export const PASSPHRASE_STORAGE_KEY = 'passphrase'
