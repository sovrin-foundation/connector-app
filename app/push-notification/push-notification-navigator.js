// @flow
import React, { PureComponent } from 'react'
import { Platform } from 'react-native'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import type {
  PushNotificationNavigatorProps,
  NextPropsPushNotificationNavigator,
  ClaimOfferPushPayload,
  AdditionalDataPayload,
  ClaimPushPayload,
} from './type-push-notification'
import {
  updatePayloadToRelevantStoreAndRedirect,
  clearNavigateToRoutePN,
} from './push-notification-store'
import type {
  NavigationParams,
  NotificationPayload,
} from '../common/type-common'
import type { Claim } from '../claim/type-claim'
import type {
  ProofRequestPushPayload,
  AdditionalProofDataPayload,
} from '../proof-request/type-proof-request'

export class PushNotificationNavigator extends PureComponent<
  PushNotificationNavigatorProps,
  void
> {
  componentWillReceiveProps(nextProps: PushNotificationNavigatorProps) {
    if (
      nextProps.pushNotification.notification &&
      nextProps.pushNotification.notification !==
        this.props.pushNotification.notification
    ) {
      this.props.updatePayloadToRelevantStoreAndRedirect(
        nextProps.pushNotification.notification
      )
    }
    if (
      nextProps.pushNotification.navigateRoute &&
      nextProps.pushNotification.navigateRoute !==
        this.props.pushNotification.navigateRoute
    ) {
      const { routeName, params } = nextProps.pushNotification.navigateRoute
      this.props.clearNavigateToRoutePN()
      this.props.navigateToRoute(routeName, params)
    }
  }

  render() {
    return null
  }
}

const mapStateToProps = ({ pushNotification, route, lock }) => ({
  pushNotification,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      updatePayloadToRelevantStoreAndRedirect,
      clearNavigateToRoutePN,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(
  PushNotificationNavigator
)
