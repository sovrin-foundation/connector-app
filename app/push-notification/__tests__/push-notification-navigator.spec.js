// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { homeTabRoute } from '../../common'
import { PushNotificationNavigator } from '../push-notification-navigator'
import {
  convertClaimOfferPushPayloadToAppClaimOffer,
  convertClaimPushPayloadToAppClaim,
  convertProofRequestPushPayloadToAppProofRequest,
} from '../push-notification-store'
import {
  claimOfferPushNotification,
  proofRequestPushPayloadAdditionalData,
} from '../../../__mocks__/static-data'

describe('<PushNotificationNavigator />', () => {
  let store = {}

  const storeToProps = {
    pushNotification: {
      notification: claimOfferPushNotification,
      navigateRoute: null,
    },
  }

  beforeAll(() => {
    store = {
      getState() {
        return {
          ...storeToProps,
        }
      },
      subscribe() {
        return jest.fn()
      },
      dispatch() {
        return jest.fn()
      },
    }
  })

  function props() {
    return {
      ...storeToProps,
      updatePayloadToRelevantStoreAndRedirect: jest.fn(),
      clearNavigateToRoutePN: jest.fn(),
      navigateToRoute: jest.fn(),
    }
  }

  it('should render PushNotification Navigator components properly', () => {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <PushNotificationNavigator {...props()} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})

describe('convertProofRequestPushPayloadToAppProofRequest', () => {
  it('should convert push payload to proof request format correctly', () => {
    expect(
      convertProofRequestPushPayloadToAppProofRequest(
        proofRequestPushPayloadAdditionalData
      )
    ).toMatchSnapshot()
  })
})

describe('convertClaimPushPayloadToAppClaim', () => {
  it('should match snapshot', () => {
    const claimPushPayload = {
      msg_type: 'claim',
      version: '0.1',
      claim_offer_id: '7TNw2k5',
      from_did: '3KFuh4jmMC5Agsy5HcCwFB',
      to_did: '5RHwxmBrGxaEskHcBnLKve',
      claim: { name: ['Test', '12'] },
      schema_seq_no: 12,
      issuer_did: 'V4SGRU86Z58d6TV7PBUe6f',
      signature: { primary_claim: { m2: 'm2', a: 'a', e: 'e', v: 'v' } },
    }
    const claim = convertClaimPushPayloadToAppClaim(
      claimPushPayload,
      '1',
      claimPushPayload.to_did
    )
    expect(claim).toMatchSnapshot()
  })
})

describe('convertClaimOfferPushPayloadToAppClaimOffer', () => {
  it('should convert push payload to claim offer format correctly', () => {
    const claimOfferPushPayload = {
      msg_type: 'CLAIM_OFFER',
      version: '0.1',
      to_did: 'BnRXf8yDMUwGyZVDkSENeq',
      from_did: 'GxtnGN6ypZYgEqcftSQFnC',
      iid: 'cCanHnpFAD',
      mid: '',
      claim: {
        name: ['Alice'],
        date_of_birth: ['2000-05-17'],
        height: ['175'],
      },
      schema_seq_no: 12,
      issuer_did: 'V4SGRU86Z58d6TV7PBUe6f',
      nonce: '351590',
      claim_name: 'Profile detail',
      issuer_name: 'Test Enterprise',
      optional_data: { terms_of_service: '<Large block of text>', price: 6 },
      remoteName: 'Test remote name',
    }

    expect(
      convertClaimOfferPushPayloadToAppClaimOffer(claimOfferPushPayload)
    ).toMatchSnapshot()
  })
})
