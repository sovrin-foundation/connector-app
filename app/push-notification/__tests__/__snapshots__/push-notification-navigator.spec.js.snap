// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PushNotificationNavigator /> should render PushNotification Navigator components properly 1`] = `null`;

exports[`convertClaimOfferPushPayloadToAppClaimOffer should convert push payload to claim offer format correctly 1`] = `
Object {
  "data": Object {
    "claimDefinitionSchemaSequenceNumber": 12,
    "name": "Profile detail",
    "revealedAttributes": Array [
      Object {
        "data": "Alice",
        "label": "name",
      },
      Object {
        "data": "2000-05-17",
        "label": "date_of_birth",
      },
      Object {
        "data": "175",
        "label": "height",
      },
    ],
    "version": "0.1",
  },
  "issuer": Object {
    "did": "V4SGRU86Z58d6TV7PBUe6f",
    "name": "Test Enterprise",
  },
  "payTokenValue": undefined,
}
`;

exports[`convertClaimPushPayloadToAppClaim should match snapshot 1`] = `
Object {
  "claim": Object {
    "name": Array [
      "Test",
      "12",
    ],
  },
  "claim_offer_id": "7TNw2k5",
  "forDID": "5RHwxmBrGxaEskHcBnLKve",
  "from_did": "3KFuh4jmMC5Agsy5HcCwFB",
  "issuer_did": "V4SGRU86Z58d6TV7PBUe6f",
  "messageId": "7TNw2k5",
  "msg_type": "claim",
  "remoteDid": "3KFuh4jmMC5Agsy5HcCwFB",
  "schema_seq_no": 12,
  "signature": Object {
    "primary_claim": Object {
      "a": "a",
      "e": "e",
      "m2": "m2",
      "v": "v",
    },
  },
  "to_did": "5RHwxmBrGxaEskHcBnLKve",
  "uid": "1",
  "version": "0.1",
}
`;

exports[`convertProofRequestPushPayloadToAppProofRequest should convert push payload to proof request format correctly 1`] = `
Object {
  "data": Object {
    "name": "proof_req_1",
    "requestedAttributes": Array [
      Object {
        "label": "Address 1",
      },
      Object {
        "label": "Address 2",
      },
    ],
    "version": "0.1",
  },
  "originalProofRequestData": Object {
    "name": "proof_req_1",
    "nonce": "123432421212",
    "requested_attributes": Object {
      "attr1_uuid": Object {
        "name": "Address 1",
      },
      "attr2_uuid": Object {
        "name": "Address 2",
      },
    },
    "requested_predicates": Object {},
    "version": "0.1",
  },
  "proofHandle": 1,
  "requester": Object {
    "name": "Evernym",
  },
}
`;
